uqcsbot/scripts/meme.py

```diff
@@ -0,0 +1,192 @@
+    "noidea": "I Have No Idea What I'm Doing",
+    "oag": "Overly Attached Girlfriend",
+    "officespace": "That Would Be Great",
+    "older": "An Older Code Sir, But It Checks Out",
+    "oprah": "Oprah You Get a Car",
+    "patrick": "Push it somewhere else Patrick",
+    "philosoraptor": "Philosoraptor",
+    "puffin": "Unpopular opinion puffin",
+    "red": "Oh, Is That What We're Going to Do Today?",
+    "regret": "I Immediately Regret This Decision!",
+    "remembers": "Pepperidge Farm Remembers",
+    "rollsafe": "Roll Safe",
+    "sad-biden": "Sad Joe Biden",
+    "sad-boehner": "Sad John Boehner",
+    "sad-bush": "Sad George Bush",
+    "sad-clinton": "Sad Bill Clinton",
+    "sad-obama": "Sad Barack Obama",
+    "sadfrog": "Sad Frog / Feels Bad Man",
+    "saltbae": "Salt Bae",
+    "sarcasticbear": "Sarcastic Bear",
+    "sb": "Scumbag Brain",
+    "scc": "Sudden Clarity Clarence",
+    "sf": "Sealed Fate",
+    "sk": "Skeptical Third World Kid",
+    "ski": "Super Cool Ski Instructor",
+    "snek": "Skeptical Snake",
+    "soa": "Seal of Approval",
+    "sohappy": "I Would Be So Happy",
+    "sohot": "So Hot Right Now",
+    "sparta": "This is Sparta!",
+    "spongebob": "Mocking Spongebob",
+    "ss": "Scumbag Steve",
+    "stew": "Baby, You've Got a Stew Going",
+    "success": "Success Kid",
+    "tenguy": "10 Guy",
+    "toohigh": "The Rent Is Too Damn High",
+    "tried": "At Least You Tried",
+    "ugandanknuck": "Ugandan Knuckles",
+    "whatyear": "What Year Is It?",
+    "winter": "Winter is coming",
+    "wonka": "Condescending Wonka",
+    "xy": "X all the Y",
+    "yallgot": "Y'all Got Any More of Them",
+    "yodawg": "Xzibit Yo Dawg",
+    "yuno": "Y U NO Guy",
+}
+
+# TODO: Would be really simple to add custom UQCS memes
+
+
+@bot.on_command("meme")
+@loading_status
+def handle_meme(command: Command):
+    """
+    !meme <meme name> "<top text>" "<bottom text>"
+
+    For a full list of names: !meme names
+    """
+    channel = command.channel_id
+
+    if not command.has_arg():
+        bot.post_message(channel, "Please run !help meme for usage")
+        return
+
+    name = command.arg.split()[0].lower()
+    if name != "names" and name not in MEME_NAMES.keys():
+        bot.post_message(channel, "The meme name is invalid. Try !meme names to get a list of all valid names")
+        return
+
+    args = get_meme_arguments(command.arg)
+
+    if name == "names":
+        send_meme_names(command)
+        return
+    elif len(args) == 2:
+        top, bottom = args
+    else:
+        bot.post_message(channel, "You supplied the wrong number of args. Please run !help meme")
+        return
+
+    # Make an attachment linking to image
+    image_url = API_URL + f"{quote(name)}/{quote(top)}/{quote(bottom)}.jpg"
+    attachments = [{"text": "", "image_url": image_url}]
+    bot.post_message(channel, "", attachments=attachments)
+
+
+@success_status
+def send_meme_names(command: Command):
```

Comment:
this appears to be the same format as sending an image. Maybe remove it from a function (which looks untidy since you're passing `command` around) and place it back into the if block?

Reply:
I took that out so I could add the @success_status unless there is an alternative way or just better to not use it?

Reply:
what do you mean so that you can add success status? you can just put it above loading status in `handle_meme` right?

Reply:
ahhhhhhhh I see-- then yeah this is probably fine, just add a comment explaining this reasoning :^)